dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java

```diff
@@ -235,8 +304,329 @@ public void Test_Constant_fields_are_in_map() throws Exception {
         
     }
 
-    
-    
+
+    @DataProvider
+    public static Object[] listTestCases() throws Exception {
+        final User user = APILocator.systemUser();
+        final ContentletAPI contentletAPI = APILocator.getContentletAPI();
+        final int limit = 300;
+        final boolean respectFrontEndUsers = false;
+        return new Object[]{
+                new TestCase(BaseContentType.getBaseContentType(0),
+                        contentletAPI.search("+basetype:0", limit, 0, null, user, respectFrontEndUsers)),
+                new TestCase(BaseContentType.getBaseContentType(1),
+                        contentletAPI.search("+basetype:1", limit, 0, null, user, respectFrontEndUsers)),
+                new TestCase(BaseContentType.getBaseContentType(2),
+                        contentletAPI.search("+basetype:2", limit, 0, null, user, respectFrontEndUsers)),
+                new TestCase(BaseContentType.getBaseContentType(3),
+                        contentletAPI.search("+basetype:3", limit, 0, null, user, respectFrontEndUsers)),
+                new TestCase(BaseContentType.getBaseContentType(4),
+                        contentletAPI.search("+basetype:4", limit, 0, null, user, respectFrontEndUsers)),
+                new TestCase(BaseContentType.getBaseContentType(5),
+                        contentletAPI.search("+basetype:5", limit, 0, null, user, respectFrontEndUsers)),
+                new TestCase(BaseContentType.getBaseContentType(6),
+                        contentletAPI.search("+basetype:6", limit, 0, null, user, respectFrontEndUsers)),
+                new TestCase(BaseContentType.getBaseContentType(7),
+                        contentletAPI.search("+basetype:7", limit, 0, null, user, respectFrontEndUsers)),
+                new TestCase(BaseContentType.getBaseContentType(8),
+                        contentletAPI.search("+basetype:8", limit, 0, null, user, respectFrontEndUsers))
+        };
+    }
+
+    private static class TestCase {
+        final BaseContentType baseContentType;
+        final List<Contentlet> contentlets;
+
+        TestCase(final BaseContentType baseContentType,
+                final List<Contentlet> contentlets) {
+            this.baseContentType = baseContentType;
+            this.contentlets = contentlets;
+        }
+    }
+
+    /**
+     * Given Scenario: We have the old and deprecated implementation vs the new one
+     * Expected Result: Both results must match
+     * @param testCase
+     */
+    @Test
+    @UseDataProvider("listTestCases")
+    public void Transformer_Backwards_Compatibility_Test(final TestCase testCase) {
+
+        final Set<String> privateInternalProperties = AbstractTransformStrategy.privateInternalProperties;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5596402)